neurodamus/utils/memory.py

```diff
@@ -513,8 +539,33 @@ def assign_cells_to_rank(rank_allocation, rank_memory, batch, batch_memory):
             all_allocation[pop] = rank_allocation
             all_memory[pop] = rank_memory
 
-        print_allocation_stats(all_allocation, all_memory)
-        export_allocation_stats(all_allocation, self._ALLOCATION_FILENAME)
+        # If cycles is not None and more than 1, distribute cells also over cycles
+        if cycles is not None and cycles > 1:
```

Comment:
`if cycles:` should be enough in that case, no?

Reply:
Mostly yes but in case someone haphazardly sets `modelbuilding_steps =1` the code executed would be basically redundant distributing cells over 1 cycle. Not a big deal TBH but also not a big deal to require a strict `cycles > 1` also for better clarity of the code.

Reply:
NVM, I've reworked that part to solve other issues I had with the shapes of the resulting dictionaries. Thanks a lot for pointing this out.